snuba/cli/entities.py

```diff
@@ -0,0 +1,59 @@
+from typing import Optional, Tuple
+
+import click
+
+from snuba.datasets.entities import EntityKey
+from snuba.datasets.entities.factory import InvalidEntityError, get_entity
+from snuba.utils.describer import Description, DescriptionVisitor
+
+
+class CLIDescriber(DescriptionVisitor):
+    def __init__(self) -> None:
+        self.__current_indentation = 0
+
+    def __indent(self) -> str:
+        return " " * self.__current_indentation
+
+    def visit_header(self, header: Optional[str]) -> None:
+        if header is not None:
+            click.echo(f"{self.__indent()}{header}")
+            click.echo(f"{self.__indent()}--------------------------------")
+
+    def visit_description(self, desc: Description) -> None:
+        self.__current_indentation += 1
+        desc.accept(self)
+        self.__current_indentation -= 1
+        click.echo("")
+
+    def visit_string(self, string: str) -> None:
+        click.echo(f"{self.__indent()}{string}")
+
+    def visit_tuple(self, tuple: Tuple[str, str]) -> None:
+        click.echo(f"{self.__indent()}{tuple[0]}: {tuple[1]}")
+
+
+@click.group()
+def entities() -> None:
+    pass
+
+
+@entities.command()
+def list() -> None:
+    """
+    Declared entities
+    """
+
+    click.echo("Declared Entities:")
+    for r in EntityKey:
+        click.echo(r.value)
+
+
+@entities.command()
+@click.argument("entity_name")
```

Comment:
I think it'd be better to make this a Choice so only valid names can get passed here and the error messages are a bit nicer. Then we won't need to try/except later

Reply:
done